rtems/Mutex.java

```diff
@@ -1,20 +1,24 @@
 package rtems;
-
 import base.Lock;
 import base.Condition;
+import java.util.Comparator;
+import java.util.PriorityQueue;
 
 public class Mutex {
 	int nestCount;
-	Object waitQueue;
-	Object holder;
-	Object orderRec; 
+	PriorityQueue<Object> waitQueue;
+	RTEMSThread holder;
+	//Object orderRec;
+	int priorityBefore; 
 	final Lock parentLock = new /*Reentrant*/Lock();
-   	final Condition cv1  = parentLock.newCondition(); 
+	final Condition cv1  = parentLock.newCondition(); 
+	MyComparator comparator = new MyComparator();
 	public Mutex()
 	{
 		this.nestCount = 0;
 		this.holder = null;
-		this.orderRec = new OrderList();
+		this.priorityBefore = -1;
+		PriorityQueue<RTEMSThread> waitQueue = new PriorityQueue<RTEMSThread>(7, comparator);
```

Comment:
stylistically I would prefer to use an anonymous class here: https://docs.oracle.com/javase/tutorial/java/javaOO/anonymousclasses.html